overlord/ifacestate/handlers.go

```diff
@@ -54,6 +54,13 @@ func (m *InterfaceManager) setupAffectedSnaps(task *state.Task, affectingSnap st
 		}
 		var snapst snapstate.SnapState
 		if err := snapstate.Get(st, affectedSnapName, &snapst); err != nil {
+			if err == state.ErrNoState {
+				// NOTE: This is a temporary measure until the root cause of issue
+				// like this can be found and corrected.
+				task.Errorf("cannot get state of snap %q that was affected by a change to snap %q -- skipping setup of its security profiles",
+					affectedSnapName, affectingSnap)
+				continue
+			}
 			return err
```

Comment:
We don't want to ever return here, no matter what the cause is. We'd again be on the same situation we're getting out of. Let's please do something like this instead:
```
task.Errorf("skipping security profiles setup for snap %q when handling snap %q: %v", affectedSNapName, affectingSnap, err)
continue
```